### ISSUE TITLE
dotnet run fails: "The specified executable is not a valid application for this OS platform"

### ISSUE THREAD

**Issue by MatthewK100000** [dotnet/sdk#30565]
### Describe the bug
I created a class library inside a folder called "PacktLibrary" using `dotnet new classlib`, and a public class called Packt.Shared within the file with the `.cs` extension. In another folder called "PeopleApp" part of the same directory as "PacktLibrary", I created a console application using `dotnet new console` referencing the class from the class library file. In the PeopleApp.csproj file I added a reference to PacktLibrary.csproj like so:
![image](https://user-images.githubusercontent.com/40904412/218533359-aa8f3688-effb-43aa-b881-2b838a767f40.png)

There are no compile time errors and everything about the syntax is correct. I have no issue running `dotnet clean`, `dotnet restore` or `dotnet build`. However, when I run `dotnet run` inside the "PeopleApp" folder I have the following issue:
![image](https://user-images.githubusercontent.com/40904412/218533901-db13cb03-a1ac-4153-a14d-f3a968171a66.png)

The dotnet CLI was installed as 64 bit, which is okay since I am running this on a 64 bit windows machine. I tried running the executable without the `dotnet` CLI tool, and I still get the same error:
![image](https://user-images.githubusercontent.com/40904412/218534717-aede3db2-3a35-4ab3-9af5-b6966a7554c1.png)

There's something wrong about the way the executable was created, perhaps a bug with the way the dotnet tool is creating the build. I'm not sure how to check the contents of the PeopleApp.exe file to reveal if it is 64 or 32 bit, using notepad doesn't work and I don't have linux so I can't run a `file ...` on the command prompt. 

### Further technical details
- Include the output of `dotnet --info`
![image](https://user-images.githubusercontent.com/40904412/218535823-de5a084b-6ba9-4204-85d2-7e2ec43e6612.png)

- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version
![image](https://user-images.githubusercontent.com/40904412/218536149-832f7b70-2e4f-4385-8cd9-97a15d88d244.png)

**Comment by joeloff** [comment]
I tried out a repro using the following commands (6.0.405 SDK on Windows Sandbox (x64):

![image](https://user-images.githubusercontent.com/7409981/219178929-1458c379-a217-4baf-80ae-302cdc00e8be.png)

Questions:
1. If you open the Windows Event Log, do you see any errors being reported?
2. Did the build actually produce an EXE
3. Did you set the `<RuntimeIdentifier>` property anywhere. For example, if I take my repro and add `<RuntimeIdentifier>linux-x64</RuntimeIdentifier>` , I see the same error as what you reported.

![image](https://user-images.githubusercontent.com/7409981/219189935-359dfd56-6765-45a7-aaa3-4870df115cf7.png)

**Comment by MatthewK100000** [comment]
Thank you for looking into this issue.

A couple of days ago I tried the above commands in a new folder `y`, and I was getting the same error message. I believe an exe file was produced as well. These were the messages I saw in Event Viewer:
![image](https://user-images.githubusercontent.com/40904412/220212808-dc209cb9-2fc8-4e9f-a289-886e1de21b14.png)
![image](https://user-images.githubusercontent.com/40904412/220212930-8b1517a0-8932-4e52-b723-fe85930b8e19.png)
![image](https://user-images.githubusercontent.com/40904412/220212949-8d489742-3f33-4c79-a444-aa2928434049.png)

So I didn't get a chance to post this until today, but then I tried running the above apps again, and viola, it works fine now. Strange!
